rpctools/jsonrpc/client.py

```diff
@@ -136,10 +140,13 @@ def __init__(self, uri, key_file=None, cert_file=None, ca_certs=None, validate_c
             auth = auth.strip()
             extra_headers.update({"Authorization": b"Basic " + auth})
 
-        self.key_file = key_file
-        self.cert_file = cert_file
-        self.ca_certs = ca_certs
         self.validate_cert_hostname = validate_cert_hostname
+        self.ssl_opts = ssl_opts or {}
+        deprecated_params = {'keyfile': key_file, 'certfile': cert_file, 'ca_certs': ca_certs}
+        for opt, val in deprecated_params.items():
+            if val is not None:
+                warnings.warn('key_file, cert_file, and ca_certs arguments are deprecated; use ssl_opts argument instead', DeprecationWarning)
+                self.ssl_opts.setdefault(opt, val)
```

Comment:
This is the actual fix.  As you can see, we're backwards-compatible with the old behavior for now, though I expect that once we tag ``1.0.0`` we'll probably eliminate the deprecated options entirely.